src/editors/containers/VideoEditor/components/VideoSettingsModal/components/DurationWidget/hooks.js

```diff
@@ -0,0 +1,219 @@
+import { useEffect, useState } from 'react';
+import { useSelector } from 'react-redux';
+
+import { actions, selectors } from '../../../../../../data/redux';
+import messages from '../messages';
+
+import * as module from './hooks';
```

Comment:
Circular dependencies can be dangerous and lead to bugs, at least they make the code very complicated. There is no need for a pattern like this, you can just declare any functions you want to use in this file in the top as `const myFunction` and use them directly and then export them below.

Reply:
We discussed this on slack but I'll layout the result of that discussion here. Refactoring out self-import should be something we take a look at in the future, but for now, I'll move ahead on this PR with self-import in play.